### PR TITLE
[UNDERTOW-2361] handle inflater wrapping properly in deflate encoding

### PR DESCRIPTION
**Pull request by baranowb**
Issue: https://issues.redhat.com/browse/UNDERTOW-2361

### CODE REVIEW

File: core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java [hunk 1]
```diff
@@ -72,18 +74,52 @@ public InflatingStreamSourceConduit(
             ObjectPool<Inflater> inflaterPool) {
         super(next);
```
Comment (ropalka) [inline core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java:75]: How about using this( ... ) constructor as above?

File: core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java [hunk 2]
```diff
@@ -171,6 +209,35 @@ public int read(ByteBuffer dst) throws IOException {
...
+        //            1      18 19   18 57   18 95   18 D3
+        //            2      28 15   28 53   28 91   28 CF
+        //            3      38 11   38 4F   38 8D   38 CB
+        //            4      48 0D   48 4B   48 89   48 C7
+        //            5      58 09   58 47   58 85   58 C3
+        //            6      68 05   68 43   68 81   68 DE
+        //            7      78 01   78 5E   78 9C   78 DA
+        buf.mark();
+        final char cmf = (char)(buf.get() & 0xFF);
+        final char flg = (char)(buf.get() & 0xFF);
```
Comment (ropalka) [inline core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java:236]: What if buf size is 1? We couldn't read 2nd byte from it then.

**Review by ropalka — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java (hunk 1 above)
Comment (baranowb) [reply core/src/main/java/io/undertow/conduits/InflatingStreamSourceConduit.java:75]: We could, but I think JDK would not like that: "The constructor InflatingStreamSourceConduit(StreamSourceConduit) is undefined"

**Review by ropalka — APPROVED** [review]

**Comment by flozano** [comment]
hi @fl4via , any idea when this would be released ?

**Comment by flozano** [comment]
hi again @fl4via , any hint when this would be released? seems the review was approved but never merged.

**Comment by fl4via** [comment]
Hi @flozano . Certainly in the next release

**Comment by flozano** [comment]
this didn't make it to 2.3.18.Final ? :(

**Comment by fl4via** [comment]
@flozano this will make it to the next release

**Comment by flozano** [comment]
thank you @fl4via

**Comment by fl4via** [comment]
Hello @flozano . This was added to 2.3.19.Final and it seems it caused a regression as you can see in #1796

**Pull request merged**